src/Cake.Testing/Extensions/ReflectionExtensions.cs

```diff
@@ -45,19 +45,29 @@ public static bool ImplementsInterfaceDefinition(this Type type, Type interfaceT
                 throw new ArgumentException("An interface type definition must be specified.", nameof(interfaceTypeDefinition));
             }
 
-            if (interfaceTypeDefinition.IsGenericType)
+            if (!interfaceTypeDefinition.IsGenericType)
             {
-                if (interfaceTypeDefinition.IsConstructedGenericType)
+                if (type == interfaceTypeDefinition)
                 {
-                    throw new ArgumentException("An interface type definition must be specified.", nameof(interfaceTypeDefinition));
+                    return true;
                 }
 
-                return type.GetInterfaces().Any(@interface =>
-                    @interface.IsGenericType
-                    && @interface.GetGenericTypeDefinition() == interfaceTypeDefinition);
+                return type.GetInterfaces().Contains(interfaceTypeDefinition);
             }
 
-            return type.GetInterfaces().Contains(interfaceTypeDefinition);
+            if (interfaceTypeDefinition.IsConstructedGenericType)
+            {
+                throw new ArgumentException("An interface type definition must be specified.", nameof(interfaceTypeDefinition));
+            }
+
+            if (type.IsGenericType && type.GetGenericTypeDefinition() == interfaceTypeDefinition)
+            {
+                return true;
```

Comment:
These are the ones we were actually missing, where the type *is* the interface rather than *implementing* the interface.